myscript.js

```diff
@@ -0,0 +1,21 @@
+
+var dataArray = [1, 8, 7, [ 4, 65, 1, [ 2, 7 ], 45, [ 6, 4], 6 ], 4, 8]
+
+var jenyaFlatternArray = function (Array) {
+    var flattering = function (mlArray) {
+        for (var i = 0; i < mlArray.length; i++) {
+            var check = typeof mlArray[i];
```

Comment:
Also, you see that you have reused line `mlArray[i]` 3 times, so let's store it in local variable and use it instead of `mlArray[i]`